### PR TITLE
Update nrf52840 generated bindings.

### PR DESCRIPTION
**Pull request by wez**
I searched crates.io for nrf52 but it didn't match your nrf52840 crate, so I made one based on the nrf52 crate, only later discovering your crate when I went to publish mine :-p

This commit uses the "best available" svd2rust to update the bindings; a number of additional peripherals are enabled.

Because I used nrf52 as a template, that's pulled in dual Apache/MIT licensing bits.  Hopefully this isn't an issue.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 documentation = "https://docs.rs/nrf52840"
-keywords = ["no-std", "arm", "cortex-m", "nrf52840"]
-license = "MIT"
+keywords = ["no-std", "arm", "cortex-m", "nrf52", "nrf52840"]
+license = "MIT OR Apache-2.0"
 name = "nrf52840"
-repository = "https://github.com/jscarrott/nrf52840"
-version = "0.1.3"
-authors = ["jscarrott <johnps@outlook.com>"]
+repository = "https://github.com/wez/nrf52840-rs"
```
Comment (wez) [inline Cargo.toml]: just noticed this; we can obviously undo the URL change

**Comment by jscarrott** [comment]
@wez When you say the best available version of svd2rust is that just the latest master version? I'll just check that my code still compiles with this version of the crate this evening and I am happy to merge it. 

crates.io not matching the crate is something that has caught me out as well, I assumed it was because it is so rarely downloaded.

**Comment by wez** [comment]
https://github.com/jamesmunns/nrf52/issues/3 is tracking the backstory and what is blocking using the released version for the nrf52 crate.  The newer version can generate code for more of the peripherals, particularly those that have overlapping address ranges, as well as correctly parse some of the fields in the newer svd file that use hexadecimal notation.

**Comment by jscarrott** [comment]
Yep that all looks good. I'll push it up to crates.io. I have half a mind to automate that bit when I have some more time.

**Pull request merged**